curp/src/server/metrics.rs

```diff
@@ -0,0 +1,178 @@
+use std::sync::{Arc, OnceLock};
+
+use clippy_utilities::NumericCast;
+use curp_external_api::{cmd::Command, role_change::RoleChange};
+use opentelemetry::{
+    global::meter_with_version,
+    metrics::{Counter, Histogram, Meter, MetricsError, ObservableGauge},
+    KeyValue,
+};
+
+use super::raw_curp::RawCurp;
+
+/// Global metrics for curp server
+static METRICS: OnceLock<Metrics> = OnceLock::new();
+
+/// Meter for metrics
+static METRICS_METER: OnceLock<Meter> = OnceLock::new();
+
+/// Get the curp metrics
+pub(super) fn get() -> &'static Metrics {
+    METRICS.get_or_init(|| Metrics::new(meter()))
+}
+
+/// Get the curp metrics meter
+fn meter() -> &'static Meter {
+    METRICS_METER.get_or_init(|| {
+        meter_with_version(
+            env!("CARGO_PKG_NAME"),
+            Some(env!("CARGO_PKG_VERSION")),
+            Some(env!("CARGO_PKG_REPOSITORY")),
+            Some(vec![KeyValue::new("component", "curp")]),
+        )
+    })
+}
```

Comment:
These code are very similar with those code define in other metric.rs files. Shall we define a macro to reduce the duplication?

Reply:
It seems a bit tricky to handle. This piece of code is shared across multiple crates, requiring a separate crate to provide the macros.